services/QuillLMS/client/app/bundles/Staff/components/comprehension/rulesAnalysis/ruleAnalysis.tsx

```diff
@@ -0,0 +1,231 @@
+    queryFn: fetchActivity
+  });
+
+  const { data: ruleFeedbackHistoryData } = useQuery({
+    queryKey: [`rule-feedback-histories-by-rule-${ruleId}`, ruleId],
+    queryFn: fetchRuleFeedbackHistoriesByRule
+  })
+
+  function handleFilterChange(e) { setFilter(e.target.value) }
+
+  function filterResponses(r) {
+    if (filter === ALL) { return true }
+    if (filter === SCORED && r.strength !== null) { return true }
+    if (filter === UNSCORED && r.strength === null) { return true }
+
+    return false
+  }
+
+   async function makeStrong(response) { updateFeedbackHistoryRatingStrength(response.response_id, true) }
+
+   async function makeWeak(response) { updateFeedbackHistoryRatingStrength(response.response_id, false) }
+
+   async function updateFeedbackHistoryRatingStrength(responseId, strong) {
+     createOrUpdateFeedbackHistoryRating({ rating: strong, feedback_history_id: responseId}).then((response) => {
+       queryCache.refetchQueries(`rule-feedback-histories-by-rule-${ruleId}`);
+     });
+   }
+
+  const ruleRows = ({ rule }) => {
+    if(!rule) {
+      return [];
+    } else {
+      // format for DataTable to display labels on left side and values on right
+      const { description, feedbacks, name, rule_type, concept_uid, } = rule;
+
+      const selectedConcept = conceptsData.concepts.find(c => c.uid === concept_uid);
+      const selectedConceptNameArray = selectedConcept ? selectedConcept.name.split(' | ') : []
+
+      const fields = [
+        {
+          label: 'API Name',
+          value: rule_type
+        },
+        {
+          label: 'Name',
+          value: name
+        },
+        {
+          label: 'Rule Description',
+          value: description ? stripHtml(description) : ''
+        },
+        {
+          label: 'Concept - Level 0',
+          value: selectedConceptNameArray[2]
+        },
+        {
+          label: 'Concept - Level 1',
+          value: selectedConceptNameArray[1]
+        },
+        {
+          label: 'Concept - Level 2',
+          value: selectedConceptNameArray[0]
+        },
+        {
+          label: 'Feedback - 1st Attempt',
+          value: feedbacks[0] ? feedbacks[0].text : null
+        },
+        {
+          label: 'Feedback - 2nd Attempt',
+          value: feedbacks[1] ? feedbacks[1].text : null
+        },
+        {
+          label: 'Responses',
+          value: ruleFeedbackHistoryData && ruleFeedbackHistoryData.responses ? ruleFeedbackHistoryData.responses.length : 0
+        }
+      ];
+      return fields.map((field, i) => {
+        const { label, value } = field
+        return {
+          id: `${label}-${i}`,
+          field: label,
+          value
+        }
+      });
+    }
+  }
+
+  // The header labels felt redundant so passing empty strings and hiding header display
+  const ruleHeaders = [
+    { name: "", attribute:"field", width: "180px" },
+    { name: "", attribute:"value", width: "1000px" }
+  ];
```

Comment:
this is just a general thought and wondering your opinion... I feel like we do this often enough that maybe we should considering updating the DataTable in a future task?  Maybe changing the header property be an optional prop and having a separate column widths property passed in when a header is unnecessary